feat (ai/ui): add reasoning support to useChat (#4513)
vercel/ai: content/docs/04-ai-sdk-ui/02-chatbot.mdx

}
```

## Reasoning

Some models such as as DeepSeek `deepseek-reasoner` support reasoning tokens.

> These tokens are typically send before the message content.

Reviewer: ```suggestion
These tokens are typically sent before the message content.
```

> You can enable forwarding them to the client with the `sendReasoning` option:

Reviewer: ```suggestion
You can forward them to the client with the `sendReasoning` option:
```

```ts filename="app/api/chat/route.ts" highlight="13"
import { deepseek } from '@ai-sdk/deepseek';
import { streamText } from 'ai';

export async function POST(req: Request) {
  const { messages } = await req.json();

  const result = streamText({
    model: deepseek('deepseek-reasoner'),
    messages,
  });

  return result.toDataStreamResponse({
    sendReasoning: true,
  });
}
```

> On the client side, you can access the reasoning text in the `reasoning` property of the message object:

Reviewer: ```suggestion
On the client side, you can access the reasoning text with the `reasoning` property on the message object:
```
